### PR TITLE
Feature/reset forgot password

### PR DESCRIPTION
**Pull request by MerciMathieu**

### CODE REVIEW

File: src/Controller/AuthController.php [hunk 1]
```diff
@@ -63,17 +69,76 @@ public function register(Request $request, EntityManagerInterface $manager, User
...
+     * @Security("is_granted('IS_ANONYMOUS')")
      */
-    public function forgotPassword()
+    public function forgotPassword(Request $request, \Swift_Mailer $mailer, UserRepository $repository, TokenGeneratorInterface $tokenGenerator, EntityManagerInterface $manager)
     {
-        return $this->render('auth/forgot-password.html.twig');
+        $form = $this->createForm(ForgotPasswordType::class);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted()) {
```
Comment (crazymad) [inline src/Controller/AuthController.php:79]: && $form->isValid() ?

File: src/Controller/AuthController.php [hunk 2]
```diff
@@ -63,17 +69,76 @@ public function register(Request $request, EntityManagerInterface $manager, User
...
+        $user = $repository->findOneBy(['resetToken' => $token]);
+        $form = $this->createForm(ResetPasswordType::class, $user);
+        $form->handleRequest($request);
+
+        if ($user && $form->isSubmitted() && $form->isValid()) {
+            $data = $form->getData();
+            $hash = $encoder->encodePassword($user, $data->getPassword());
+            $user->setPassword($hash);
+            $user->setResetToken(null);
+            $manager->persist($user);
```
Comment (crazymad) [inline src/Controller/AuthController.php:132]: no need to persist, the user is already in the database

**Review by crazymad — CHANGES_REQUESTED** [review]

File: src/Controller/AuthController.php (hunk 1 above)
Comment (MerciMathieu) [reply src/Controller/AuthController.php:79]: done

File: src/Controller/AuthController.php (hunk 2 above)
Comment (MerciMathieu) [reply src/Controller/AuthController.php:132]: understood

**Pull request merged**